### DISCUSSION TITLE
How about removing ZAP?

### DISCUSSION THREAD

**Discussion by jptosso** [corazawaf/coraza#121] [General]
Zap is a huge library and it requires a lot of dependencies, I'm not sure if it is worth it to keep using zap. 

- Are the alternatives, like logrus, faster?
- How many indirect dependencies would we remove?
- How much would be the size loss of a Coraza package?
- Is structured logging really faster than "concatenating strings"?
- Are there similar, better alternatives?

**Comment by nacx** [comment]
Another option is to consider something like https://github.com/tetratelabs/telemetry/blob/master/logger.go which is just an interface. This has zero dependencies and users would be free to choose their implementation. If there are other interface driven libraries, then that would be great.

**Reply by jptosso** [reply]
Hey @nacx, thank you for your comment. ZAP was already removed from v3 and replaced with a similar logging interface that can be easily replaced. https://github.com/corazawaf/coraza/pull/337

**Comment by jcchavezs** [comment]
We moved away from zap in v3 using std logger. We can close this discussion
now.
